spec/model_hooks_spec.rb

```diff
@@ -34,33 +29,65 @@
         Copyable::ModelHooks.disable!(CopyableTree)
         Copyable::ModelHooks.reenable!(CopyableTree)
         expect {
-          tree = CopyableTree.create!(kind: 'magnolia')
+          CopyableTree.create!(kind: 'magnolia')
         }.to raise_error(RuntimeError, "callback2 called")
       end
     end
   end
 
-  context 'observers' do
+  context 'validations' do
 
     # Note: the relevant model and observer class is defined in helper/test_models.rb
 
     describe '.disable!' do
       before(:each) do
-        Copyable::ModelHooks.disable!(CopyableCar)
+        Copyable::ModelHooks.disable!(CopyableCoin)
       end
       after(:each) do
-        Copyable::ModelHooks.reenable!(CopyableCar)
+        Copyable::ModelHooks.reenable!(CopyableCoin)
       end
-      it 'should prevent observers from executing' do
+      it 'should prevent validations from executing' do
         expect {
-          car = CopyableCar.create!(make: 'Ferrari', model: 'California', year: 2009)
+          CopyableCoin.create!(year: -10)
         }.to_not raise_error
       end
       it 'should not prevent model actions from executing' do
-        expect(CopyableCar.count).to eq(0)
-        car = CopyableCar.create!(make: 'Ferrari', model: 'California', year: 2009)
-        expect(CopyableCar.count).to eq(1)
+        expect(CopyableCoin.count).to eq(0)
+        CopyableCoin.create!(year: -10)
+        expect(CopyableCoin.count).to eq(1)
       end
     end
+
+    describe '.reenable!' do
+      it 'should allow validations to execute again' do
+        Copyable::ModelHooks.disable!(CopyableCoin)
+        Copyable::ModelHooks.reenable!(CopyableCoin)
+        expect {
+          CopyableCoin.create!(year: -10)
+        }.to raise_error
+      end
+    end
+  end
+
+  describe 'nested disables and enables' do
+    it 'should allow callbacks to execute again' do
+      Copyable::ModelHooks.disable!(CopyableTree)
+      expect { CopyableTree.create!(kind: 'magnolia') }.to_not raise_error
+
+      Copyable::ModelHooks.disable!(CopyableCoin)
+      expect { CopyableCoin.create!(year: -10) }.to_not raise_error
+
+      Copyable::ModelHooks.disable!(CopyableTree)
+      expect { CopyableTree.create!(kind: 'magnolia') }.to_not raise_error
+
+      Copyable::ModelHooks.reenable!(CopyableCoin)
+      expect { CopyableCoin.create!(year: -10) }.to raise_error
```

Comment:
I think we should be specific about what kind of Error we expect this to raise (i.e. `expect { CopyableCoin.create!(year: -10) }.to raise_error(SomeError)`. Otherwise the tests could pass due to an unrelated error.

Reply:
Will do!